docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,531 @@
+
+### Using Fetch
+
+Here's an example of how to call the `simulateTransaction` endpoint directly using `fetch` in JavaScript:
+
+```javascript
+async function simulateTransaction(transactionXDR) {
+  const requestBody = {
+    jsonrpc: "2.0",
+    id: 8675309,
+    method: "simulateTransaction",
+    params: {
+      transaction: transactionXDR,
+      resourceConfig: {
+        instructionLeeway: 50000,
+      },
+    },
+  };
+
+  const response = await fetch(
+    "https://soroban-testnet.stellar.org/simulateTransaction",
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(requestBody),
+    },
+  );
+
+  const result = await response.json();
+  console.log(result);
+}
+
+// Example XDR transaction envelope
+const transactionXDR =
+  "AAAAAgAAAAAg4dbAxsGAGICfBG3iT2cKGYQ6hK4sJWzZ6or1C5v6GAAAAGQAJsOiAAAAEQAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAGAAAAAAAAAABzAP+dP0PsNzYvFF1pv7a8RQXwH5eg3uZBbbWjE9PwAsAAAAJaW5jcmVtZW50AAAAAAAAAgAAABIAAAAAAAAAACDh1sDGwYAYgJ8EbeJPZwoZhDqEriwlbNnqivULm/oYAAAAAwAAAAMAAAAAAAAAAAAAAAA="; // Replace with your actual XDR transaction envelope
+simulateTransaction(transactionXDR);
+```
+
+### Using the JavaScript SDK
+
+The Stellar SDK provides a convenient method to simulate a transaction:
+
+```javascript
+const StellarSdk = require("stellar-sdk");
+
+// Initialize the server
+const server = new StellarSdk.Server("https://soroban-testnet.stellar.org");
+
+// Define the transaction
+const transaction = new StellarSdk.TransactionBuilder(account, {
+  fee: StellarSdk.BASE_FEE,
+})
+  .setNetworkPassphrase(StellarSdk.Networks.TESTNET)
+  .setTimeout(30)
+  .addOperation(
+    StellarSdk.Operation.invokeHostFunction({
+      function: "increment",
+    }),
+  )
+  .build();
+
+server.simulateTransaction(transaction).then((sim) => {
+  console.log("cost:", sim.cost);
+  console.log("result:", sim.result);
+  console.log("error:", sim.error);
+  console.log("latestLedger:", sim.latestLedger);
+});
+```
+
+## Understanding the Footprint
+
+A footprint is a set of ledger keys that the transaction will read or write. These keys are marked as either read-only or read-write:
+
+- **Read-Only Keys**: Available for reading only.
+- **Read-Write Keys**: Available for reading and writing.
+
+The footprint ensures that a transaction is aware of all the ledger entries it will interact with, preventing unexpected errors during execution.
+
+## Assembling a Transaction
+
+Once you have simulated the transaction and obtained the necessary data, you can assemble the transaction for actual submission. The `assembleTransaction` function in the SDK helps with this process, but you can also call `prepareTransaction` to have it both simulate and assemble the transaction for you.
+
+### javascript-sdk-assembleTransaction
+
+Here's how the `assembleTransaction` function in the javascript sdk works to combine the raw transaction with the simulation results:
```

Comment:
Again, this should be a cross-reference alongside an explanation rather than a copy-pasta in case it ever changes.

Reply:
I added a cross reference and deleted the copied code.